### DISCUSSION TITLE
Text Box Extraction

### DISCUSSION THREAD

**Discussion by gayathri-pan** [pymupdf/PyMuPDF-Utilities#134] [Q&A] [answered]
[input.pdf](https://github.com/pymupdf/PyMuPDF-Utilities/files/15454761/input.pdf)
i have to extract the first page text from the this above pdf. i used [multi_column.py](https://github.com/pymupdf/PyMuPDF-Utilities/blob/master/text-extraction/multi_column.py) file and ran without error and i have to extract the text from the rectangle. the rectangle is not recognizing. for text extraction i used  textbox-extract-2.py. kindly help me.

**Comment by JorjMcKie** [comment]
Please be more specific: what did you and what did go wrong!

**Comment by gayathri-pan** [comment]
Hello, i actually resolved the error. i facing another problem. My pdf first page have two column and in the right side column have table. does the [multi_column.py](https://github.com/pymupdf/PyMuPDF-Utilities/blob/master/text-extraction/multi_column.py) code recognize the table?

**Reply by JorjMcKie** [reply, marked as answer]
No.
Please use the new package [pymupdf4llm](https://pypi.org/project/pymupdf4llm/) which recognizes columns and tables producing Markdown text output.
